### DISCUSSION TITLE
Welcome to Pandora Behaviour Engine+ Discussions!

### DISCUSSION THREAD

**Discussion by Monitor221hz** [Monitor221hz/Pandora-Behaviour-Engine-Plus#34] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
  
  A reminder that all **pictures, videos, and other visual media must remain SFW** as per Github's acceptable use policies. Violations of this will result in sanctions.

**Comment by DKLYNTLY** [comment]
when ever i run pandora and press depoly, it asks to delete a bunch of behavior files

**Reply by Monitor221hz** [reply]
it's probably replacing existing files, so yes

**Comment by DKLYNTLY** [comment]
after i run pandora the mod manager asks to delete a bunch of skeleton.hkx files from character assests of vamipre lords and beasts do i revert the changer or delete the files?

**Comment by Tyrthemis** [comment]
Testing in VR, but I see those dll files and it worries me.  Should I expect it to work? Also is there a readme file anywhere? When it says launched in 0.62 seconds 14000 animations added, does that mean its done?

**Reply by Monitor221hz** [reply]
yes, yes, yes

**Comment by ABakedWaffle** [comment]
I'm using XP32, and I was told there was a patch for it on here, but I'm having trouble find it. Is there one?

**Reply by Monitor221hz** [reply]
https://github.com/Monitor144hz/Pandora-Behaviour-Engine-Plus/releases/tag/v1.1.3-alpha

Get the patch files only from this release

**Reply by Tyrthemis** [reply]
Is this a dummy file, kind of like Nemesis had for FNIS?  I still don't need FNIS or Nemesis installed, correct?  BTW, working well in VR so far, a few creature animations I've been struggling with might just be me figuring out creature animations for the first time.

**Reply by Monitor221hz** [reply]
It is a patch file for the skeleton to blend properly during blocking and magic usage. I'll merge it in with the main release later.

**Reply by tim-elmer** [reply]
Looks like this was done in #127, so we no longer need the separate patch, correct?

**Reply by Tyrthemis** [reply]
> Looks like this was done in #127, so we no longer need the separate patch, correct?

Is it working well @tim-elmer ? I just upgraded from 1.5.1a, was wondering if I should still be using the patch

**Reply by tim-elmer** [reply]
I haven't noticed any issues.

**Comment by ghost** [comment]
Hello,

I was only curious since there is no clear instruction/information about this or I simply couldn't find it.
Is Pandora meant to be a standalone engine or do I have to use Nemesis as a base?

**Reply by Tyrthemis** [reply]
Yes, i’ve been using it standalone, and it works just fine as far as I can tell. you  can do human animations, as well as creature animations all in one

**Reply by ghost** [reply]
Are you by chance using OCPA NG as a Power Attack supplementation?

**Reply by Tyrthemis** [reply]
I don’t know what that is.  I’m also in VR I think all I have for power attacks is PLANCK

**Reply by ghost** [reply]
Thanks a lot, much appreciated! :)

**Comment by blippyp** [comment]
Thankyou so much for making this! I hate Nemesis, always have. took me over a year to actually see it work in my mod list, took forever to figure out that something as common as Devious Devices was stopping it from working, then I had to go through each mod one by one to determine what I needed to get rid of to use it. Today Nemesis decided to take a massive dump on my load order again and refused to run, half the time I have to run it like 10 times until it 'eventually' works - It's a damn guessing game using that piece of garbage.

Frustrated after about the 50'th time running it and even after disabling any mods I just installed and rebooted my machine still without any luck with this NEMESIS garbage - NEMESIS IS MY NEMESIS (its name is ironic), I looked around and was shocked when I read someone mentioning Pandora on a reddit. I installed it, ran it, It JUST WORKS.

I absolutely love everyone contributing to this project, please keep up the good work. Saviours! everyone working on this thing and helping with it, ABSOLUTE HERO'S. I'm sure it has issues regarding something, but for me it just worked. I just started modding again recently, so my mod list isn't overly complicated, but it's far from simple also. I can BREATH again, it's as effortless as the old days with using FNIS. THANKYOU THANKYOU THANKYOU!

**Comment by Tyrthemis** [comment]
Is there a read me I can reference? Hey peeps, this might be related to the mod but I don't know for sure.  My skeletons are "A posing" and not attacking, but they will die and fall apart if I hit them, every other NPC seems to be working fine, I've had a few floating (and flying up and away instead of walking) mammoths occasionally though. to complicate things I installed a skeleton mod (https://www.nexusmods.com/skyrimspecialedition/mods/52845).  I don't have it anymore and have started a new game, but I'm wondering if installing it and uninstalling it while using pandora may have something to do with it.  If anyone has advice I would be grateful.  I've heard Bethesda reuses the draugr anims for skeletons, is this true?  Also in my data/meshes/actors/ folder, should there be a folder named skeleton? That mod put a file in there called testskeleton.hkx which I think may be the issue.  While troubleshooting I have deleted the folder as well as verified my files on steam, which didn't seem to help.  Also if we do need that folder, should it have behaviors, animations, character, and character assets folders within it?  THANK YOU for reading all that and thank you for any help.

**Reply by Tyrthemis** [reply]
I tried switching back to my nemesis profile but that didn't seem to help, which leds me to believe I'm missing files or something

**Comment by Michael-wigontherun** [comment]
It says that conflicts are rare, and you can drag and drop to overwrite. But what if there is a conflict, will it report that information?

**Comment by Tyrthemis** [comment]
It seems to not work with CGO stripped, which worked fine for Nemesis.  More details: the check box pops up when I install CGO stripped, and it looks like the original mod to Nemesis and Pandora, I can check the box and run the engine, but with Pandora, I am not getting those effects in game.  Mainly, I can't continue to attack while jumping.

**Reply by Tyrthemis** [reply]
Update: I did a clean install and intalled nemesis and all those mods worked fine, so it may be a Pandora issue.

**Reply by Monitor221hz** [reply]
@Tyrthem if you think it's a Pandora issue please submit a bug report in the Issues tab.

**Comment by VredemanH** [comment]
i can't believe it... you're... dragonborn! it actually works, been trying fnis and nemesis, but something always goes wrong, i don't know anything about modding, scrippting, programing, just made this account to thank Monitor144hz and SARDONYX-sard, i will be the first to give kudos and endorse on nexus.
THANK YOU!

**Reply by VredemanH** [reply]
i've been testing pandora and its amazing, fast, robust and most important, fool-proof. good work!
really sad that doesn't work with zaz... but eventually will... right? please?

**Comment by SarituaAlbert** [comment]
So ive been wondering, whats the animation limit for Pandora? Is there any .ini file to increase it just like Nemesis?

**Reply by SarituaAlbert** [reply]
Also, FNIS have this patch for the XXXXL version back then to increase the animation limit to 100k+. Already tried it and it works fine for my FNIS before. I wonder if this Pandora gonna have similiar things/settings to increase the animation limit.

**Reply by Monitor221hz** [reply]
The behaviour engine has no limit. The game engine does.

**Reply by SarituaAlbert** [reply]
Hmm i see, so i have this alert before, something like "Reaching" animation limit 16900/16000 (Not the exact number) that coming back after using Pandora and all my ADXP-MCO & Combat Animation just stop working after that. I do have Animation Limit Crash SSE and SSE Engine Fixes, but it doesnt fix anything. Any idea?

**Reply by John95ac** [reply]
I assume that you are using DAR, you can modify your limit by entering the folder where you installed it and looking for an ini or txt in which the limit is 16k, but I do not recommend it at all since DAR will give you random ctd, I recommend OAR (or you already have it installed and have not modified the limit within the game) and when you install it and you are in the game, press Shift + O then press settings then at the end of the new window you just press the increase animation limits box, in general settings you increase the number of animations from 16k to whatever you want you can up to 64k. (I want to clarify that these 64k are not the limit that Pandora can install, I have 100k animations and I have the limit in OAR at 64k and the screen does not appear)

**Reply by SarituaAlbert** [reply]
There is no .ini or .txt inside DAR, its just .dll but ill give a try for OAR. IDK about the CTD from DAR since i havent encounter one. But thanks for the info.

**Reply by SarituaAlbert** [reply]
Okay took me awhile to learn OAR ingame menu and i am able to configure some things. While doing this i notice that some of the animation from "Immersive Interaction SSE" like lockpicking/opening doors is not working with Pandora, is this a known issues?

**Comment by Kekkening** [comment]
I'm having an issue, installed it, added as an exe in MO2, created an output folder for it as it was instructed, ticked all the mod boxes,it created 41k animations, the output mod is active BUT when I load into the game no animations are loading (I'm used to watch the counter go up while using other engines) and my character is just T or more like A posing. What am I missing or screwed up?

I'm ticking the 3 true directional movement mods, the xpmse patch and the pandora base options.

I'm using a wabbajack-ed version of skyrim 1.5.97

![kép_2024-04-21_190806508](https://github.com/Monitor144hz/Pandora-Behaviour-Engine-Plus/assets/167704042/0bc7243c-48c5-4075-afb8-29e1a709b568)

**Reply by Monitor221hz** [reply]
@Kekkening you'll have to post the output files in a bug report for me to take a proper look, I can't tell what's wrong.

**Reply by Kekkening** [reply]
I gave up sooner and reinstalled everything, the wabba-pack, and mods. I read it (maybe it was for the earlier builds) that Pandora had issues with ZaZ, so I avoided it. This time everything went buttery smooth. Anyway thank you for offering help and for this amazing tool! Well done!

**Comment by John95ac** [comment]
Thanks for Pandora, I used Nemesis and FNIS but I always had unexpected errors due to the 100k animations I have, I thank you very much as soon as I have a stable job I will support as I can, but first I have an observation, the animations for ZAZ 8+ do not seem to work, any 
ideas?

**Reply by John95ac** [reply]
https://github.com/Monitor144hz/Pandora-Behaviour-Engine-Plus/issues/56 press 56 there are some discoveries I found in relation to ZAZ and Pandora.

**Comment by John95ac** [comment]
Hello, sorry for so many messages, but I found the solution for ZAZ 8+ to work, at least a procedure for it to work with Pandora for the moment until it is natively compatible. https://github.com/Monitor144hz/Pandora-Behaviour-Engine-Plus/issues/56 Tomorrow I will publish a pdf with the procedure and the patch. in addition to a brief explanation of how it is done.

**Reply by John95ac** [reply]
[Pandora+ZAZ patch method ES.pdf](https://github.com/Monitor144hz/Pandora-Behaviour-Engine-Plus/files/15155188/Pandora%2BZAZ.patch.method.ES.pdf)
[Pandora+ZAZ patch method EN.pdf](https://github.com/Monitor144hz/Pandora-Behaviour-Engine-Plus/files/15155190/Pandora%2BZAZ.patch.method.EN.pdf)
I leave the method in English and Spanish to patch ZAZ on Pandora. thanks for the engine. There is more information on the zaz bug page

**Comment by zhuabaobao123** [comment]
This mod is great, can you post it on the n network, this site is not used to finding something

**Reply by Monitor221hz** [reply]
> This mod is great, can you post it on the n network, this site is not used to finding something

I will post on Nexus in due time.

**Reply by zhuabaobao123** [reply]
INFO : Skyrim Patcher 1.4.2-alpha
WARN : Dispatcher > "Pandora Base" > dragonproject~dragonbehavior > Replace > Element > #0885/legs/Element0/maxAnkleHeightMS > FAILED
WARN : Dispatcher > "Pandora Base" > dragonproject~dragonbehavior > Replace > Element > #0885/legs/Element0/hipIndex > FAILED
WARN : Dispatcher > "Pandora Base" > dragonproject~dragonbehavior > Replace > Element > #0885/legs/Element0/kneeIndex > FAILED
WARN : Dispatcher > "Pandora Base" > dragonproject~dragonbehavior > Replace > Element > #0885/legs/Element0/ankleIndex > FAILED
WARN : Dispatcher > "Pandora Base" > dragonproject~dragonbehavior > Replace > Element > #0885/legs/Element0/isPlantedMS > FAILED
WARN : Dispatcher > "BFCO - Attack Behavior" > defaultfemale~1hm_behavior > Remove > Element > #3108/triggers/Element2 > FAILED
WARN : Dispatcher > "BFCO - Attack Behavior" > defaultfemale~1hm_behavior > Remove > Element > #3108/triggers/Element4 > FAILED
WARN : Dispatcher > "True Directional Movement - 360 Horse Archery" > defaultfemale~horsebehavior > Replace > Text > #0742/errorOutTranslation > FAILED
WARN : Validator > defaultfemale~1hm_behavior > Duplicate Event > Collision_Cancel > Index > 0 > REMOVED
INFO : Pandora Mod 1 : BFCO - Attack Behavior - v.1.0.0
  

Excuse me, how should I handle these few

**Comment by zhuabaobao123** [comment]
Hi, I'm here again. Can you add a Gender Specific Animations? The Princess Horse Riding mod requires it.
2. Elden Rim - Weapon Arts 3.2.8   The dodge function of this mod is not working properly. I have already used the TK Dodge Animation Pandora Patch, but it still doesn't work

**Comment by schemingl** [comment]
Pandora works mostly great so far as a successor to both fnis and nemesis. I'm so glad I can play with cool mco animations and a certain lab's creature animations at the same time.
As for the possible bugs I found (not sure if Pandora but I only got these after switching to Pandora):
- At least 2 animations from the mod Immersive Interactions don't play for the player (the "open door" and "lockpicking" animations)
- At least 1 vanilla animation doesn't play, namely Soul Cairn's skeletons emerging from the ground, leading to them being stuck in A-pose.

**Comment by VredemanH** [comment]
you must be some wizard in a secret lair doing some crazy stuff with programing magic or something like that, at least is what comes in my mind, great job!

**Reply by Monitor221hz** [reply]
Funny that you say that, you can read the Performance Notes page on the wiki if you want to know how the cake is baked. It's not magic, just a lot of nights spent typing away 🥲

**Comment by GelidaTerram** [comment]
Good day! First time trying Pandora, it runs outside of MO2 with no problem (except it only has Pandora and the Skeleton as options. Trying to run it in MO2 results in a crash every time. I reinstalled the Desktop framework to be sure, didn't help. Launched from MO2 the window opens with no mods in it, spinning wheel for a second, crash to desktop. I looked on Google but I couldn't find anything beyond some conversations about Vortex issues. Anyone care to weigh in?

**Reply by GelidaTerram** [reply]
NVM. Antivirus caused it. I forgot to add an exemption. Sorry folks!
